src/pages/SampleListPage.tsx

```diff
@@ -53,16 +56,41 @@ export const SampleListPage = () => {
     }));
   }, [data]);
 
+  useEffect(() => {
+    if (selectionModel.length > 0 && rows) {
+      let selected = selectionModel.map((i: number) => {
+        return rows[i].accession;
+      });
+      setAccessions({ identifier: { accession: selected } });
+    }
+  }, [selectionModel, rows]);
+
   if (!rows) {
     return <>Loading...</>;
   }
 
   return (
     <>
       <PageHeaderWithReturn title='Selected samples' to={`../explore${searchString}`} />
+      {selectionModel.length > 0 && (
+        <div className='m-8 flex flex-row flex-wrap'>
+          <div className='mx-4 my-1' style={{ zIndex: 10 }}>
+            <TopButtons selector={accessions} hideTaxonium={true} hideSequenceTableButton={true} />
+          </div>
+        </div>
+      )}
 
       <div style={{ width: '100%' }}>
         <DataGrid
+          checkboxSelection
+          selectionModel={selectionModel}
+          onSelectionModelChange={newSelection => {
+            let res = [];
+            for (let i of newSelection) {
+              res.push(i);
+            }
```

Comment:
Why is it needed to copy the array? If it is needed to copy, can this be shortened to `const res = [...newSelection];`?